compiler/src/iree/compiler/Codegen/Common/TransformExtensions/CommonExtensions.cpp

```diff
@@ -568,6 +569,138 @@ void transform_dialect::CopyTensorOperandOp::getEffects(
   transform::modifiesPayload(effects);
 }
 
+//===---------------------------------------------------------------------===//
+// FlattenForallOp
+//===---------------------------------------------------------------------===//
+
+template <typename MappingTy>
+bool isAscendingRelativeMapping(ArrayRef<Attribute> mapping) {
+  auto start = cast<MappingTy>(*mapping.begin());
+  if (start.getMappingId() !=
+      static_cast<int64_t>(gpu::MappingId::LinearDim0)) {
```

Comment:
Do we need this check here? Is there a reason that it should fail if using dimX, dimY, dimZ? This also seems specific to `gpu::MappingId`, but the function is templated.

Reply:
It is templated because thread and warp mapping attributes expose the same `getMappingId` API that I want to query. We could add support for x/y/z in the future, but I don't see an immediate case for using anything other than relative mappings right now (they quickly stop working for higher dimensional ops).